pkg/kubectl/resource_printer.go

```diff
@@ -154,12 +195,23 @@ func (p *JSONPrinter) PrintObj(obj runtime.Object, w io.Writer) error {
 // The input object is assumed to be in the internal version of an API and is converted
 // to the given version first.
 type YAMLPrinter struct {
-	version   string
-	convertor runtime.ObjectConvertor
+	preprocessor ObjectPreprocessor
+}
+
+func (p *YAMLPrinter) AddObjectPreprocessor(newPreprocessor ObjectPreprocessor) {
+	p.preprocessor = addObjectPreprocessor(p.preprocessor, newPreprocessor)
 }
 
 // PrintObj prints the data as YAML.
 func (p *YAMLPrinter) PrintObj(obj runtime.Object, w io.Writer) error {
+	if p.preprocessor != nil {
```

Comment:
If this were an `ObjectPreprocessors` as described above, no nil check is needed.